internal/commands/vorpal-engine.go

```diff
@@ -0,0 +1,125 @@
+package commands
+
+import (
+	"path/filepath"
+
+	"github.com/checkmarx/ast-cli/internal/commands/util/printer"
+	errorConstants "github.com/checkmarx/ast-cli/internal/constants/errors"
+	commonParams "github.com/checkmarx/ast-cli/internal/params"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func runScanVorpalCommand() func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		vorpalLatestVersion, _ := cmd.Flags().GetBool(commonParams.VorpalLatestVersion)
+		sourceFlag, _ := cmd.Flags().GetString(commonParams.SourcesFlag)
+
+		scanResult, err := ExecuteVorpalScan(sourceFlag, vorpalLatestVersion)
+		if err != nil {
+			return err
+		}
+
+		err = printer.Print(cmd.OutOrStdout(), scanResult, printer.FormatJSON)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func ExecuteVorpalScan(sourceFlag string, vorpalUpdateVersion bool) (*ScanResult, error) {
+	if sourceFlag == "" {
+		return nil, errors.Errorf(errorConstants.FileSourceFlagIsRequired)
+	}
+
+	if filepath.Ext(sourceFlag) == "" {
+		return nil, errors.New(errorConstants.FileExtensionIsRequired)
+	}
+
+	if vorpalUpdateVersion {
+		return ReturnSuccessfulResponseMock(), nil
+	}
+	return ReturnFailureResponseMock(), nil
+}
+
+func ReturnSuccessfulResponseMock() *ScanResult {
+	return &ScanResult{
+		RequestID: "1234567890",
+		Status:    true,
+		Message:   "Scan completed successfully.",
+		ScanDetails: []ScanDetail{
+			{
+				Language:    "Java",
+				QueryName:   "SQL Injection",
+				Severity:    "High",
+				FileName:    "Main.java",
+				Line:        42,
+				Remediation: "Use prepared statements or parameterized queries to prevent SQL injection.",
+				Description: "This vulnerability allows an attacker to execute malicious SQL queries.",
+			},
+			{
+				Language:    "JavaScript",
+				QueryName:   "Cross-Site Scripting (XSS)",
+				Severity:    "Medium",
+				FileName:    "index.js",
+				Line:        27,
+				Remediation: "Escape or sanitize user input before rendering it in HTML to prevent XSS attacks.",
+				Description: "This vulnerability allows an attacker to inject malicious scripts into web pages viewed by other users.",
+			},
+		},
+	}
+}
+
+func ReturnFailureResponseMock() *ScanResult {
+	return &ScanResult{
+		RequestID: "1111",
+		Status:    false,
+		Message:   "Scan failed.",
+		Error:     &Error{InternalError, "An internal error occurred."},
+	}
+}
+
+type ScanRequest struct {
+	ID         string `json:"id"`
+	FileName   string `json:"file_name"`
+	SourceCode string `json:"source_code"`
+}
+
+type SingleScanRequest struct {
+	ScanRequest ScanRequest `json:"scan_request"`
+}
+
```

Comment:
@OrShamirCM 
I think those structs may cause conflicts with the proto struct.
It may be better to merge my PR first and then use the proto structs instead this one

Reply:
@AlvoBen I removed the request as it's not relevant to map to this struct.
But about the Response - it should be similar as I took it from the proto. Let's discuss it.

Reply:
Discuss with Ben

Reply:
Discussed